gm4_mountaineering/data/gm4_mountaineering/functions/climbing_effects/kill_shulkers.mcfunction

```diff
@@ -0,0 +1,7 @@
+# kill climbing shulkers
+# @s = player wearing crampons
+# at @s
+# run from climbing_effects/damage_shulkers and climbing_effects/clear_climbing_effects
+
+teleport @e[type=minecraft:shulker, tag=gm4_mountaineering_kill_shulker] ~ ~-400 ~
+kill @e[type=minecraft:shulker, tag=gm4_mountaineering_kill_shulker]
```

Comment:
This function should execute as the shulker that needs to be killed, so you don't need the `gm4_mountaineering_kill_shulker` tag and you can use `@s` instead.